### PR TITLE
#29 Feature: Admin disable account

### PR DESCRIPTION
**Pull request by ALVINdimpos**
#### What does this PR do?
This PR adds the functionality for the admin to disable user accounts for various reasons, providing the option to block usage of a particular account and send an email to the user with the reasons for the block.

#### Description of Task to be completed?
The task involves implementing the necessary functionality in the admin dashboard to list all users and provide the option to disable their accounts. Additionally, an email notification should be sent to the user when their account is disabled, specifying the reasons for the block.

#### How should this be manually tested?
```
git clone https://github.com/atlp-rwanda/ecommerce-app-predators-fe.git
cd ecommerce-app-predators-fe
git checkout ft/admin-disable-acount-29
npm i
npm run dev
```

#### What are the relevant pivotal tracker/Trello stories?
Trello [Link](https://trello.com/c/Wsu4lTyE/29-admin-should-be-able-to-disable-an-account-for-various-reasons) (#-29)

#### Screenshots (if appropriate)

### CODE REVIEW

**Review by faid-terence — APPROVED** [review]
Good Job @ALVINdimpos

**Review by kagaba-etienne — APPROVED** [review]
Good Job!

**Review by NTElissa — APPROVED** [review]
good progress

File: src/components/table/UserTable.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+   * @type {function}
+   */
+  const dispatch = useDispatch();
+   /**
+   * Array of user objects fetched from the Redux store.
+   * @type {User[]}
+   */
+  const users = useSelector((state: any) => state.getAllUsers.data);
+  useEffect(() => {
+    dispatch(getAllUsers() as any);
```
Comment (chanceDira) [inline src/components/table/UserTable.tsx:28]: can you use interface here,

**Review by chanceDira — DISMISSED** [review]
Nice work @ALVINdimpos 
Just few issues

**Review by chanceDira — CHANGES_REQUESTED** [review]
Create interfaces, Let's not use any

**Review by nili-alu — APPROVED** [review]
great job

**Review by vanessa-abewe — APPROVED** [review]

**Review by chanceDira — APPROVED** [review]
Nice work

**Pull request merged**